docs/reference/esql/functions/types/match.asciidoc

```diff
@@ -5,8 +5,33 @@
 [%header.monospaced.styled,format=dsv,separator=|]
 |===
 field | query | result
+boolean | boolean | boolean
+boolean | keyword | boolean
+date | date | boolean
+date | keyword | boolean
+date_nanos | date_nanos | boolean
+date_nanos | keyword | boolean
+double | double | boolean
+double | integer | boolean
+double | keyword | boolean
+double | long | boolean
+integer | double | boolean
+integer | integer | boolean
+integer | keyword | boolean
+integer | long | boolean
+ip | ip | boolean
+ip | keyword | boolean
 keyword | keyword | boolean
-keyword | text | boolean
+long | double | boolean
+long | integer | boolean
+long | keyword | boolean
+long | long | boolean
 text | keyword | boolean
-text | text | boolean
+unsigned_long | double | boolean
+unsigned_long | integer | boolean
+unsigned_long | keyword | boolean
+unsigned_long | long | boolean
+unsigned_long | unsigned_long | boolean
+version | keyword | boolean
+version | version | boolean
```

Comment:
this looks great - it's exactly what we need - is this the exact table that gets generated when we run the match tests or did you have to do any changes?
I haven't looked closely at how this gets generated, but I assume that because we have the `checkParamCompatibility` check then we only get the valid combinations here?

Reply:
This is automatically generated from the `MatchTests`, based on the parameters provided as part of the `parameters()` method.

The test generates the positive test cases, and then uses the `errorsForCasesWithoutExamples` method to add failing tests for the remaining combinations. These failing tests include the tests that will fail the `checkParamCompatibility` condition.